Tests/xSharePoint/xSharePoint.xSPAddInManagementServiceApp.Tests.ps1

```diff
@@ -0,0 +1,120 @@
+            DatabaseName = "Test_DB"
+            DatabaseServer = "TestServer\Instance"
+        }
+        Import-Module (Join-Path ((Resolve-Path $PSScriptRoot\..\..).Path) "Modules\xSharePoint")
+
+        
+        Mock Invoke-xSharePointCommand { 
+            return Invoke-Command -ScriptBlock $ScriptBlock -ArgumentList $Arguments -NoNewScope
+        }
+        
+        Import-Module $Global:CurrentSharePointStubModule -WarningAction SilentlyContinue
+
+        Context "When no service applications exist in the current farm" {
+
+            Mock Get-SPServiceApplication { return $null }
+            Mock New-SPAppManagementServiceApplication {  return  @(@{})}
+            Mock New-SPAppManagementServiceApplicationProxy{ return $null }
+
+            It "returns null from the Get method" {
+                Get-TargetResource @testParams | Should BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+            It "returns false when the Test method is called" {
+                Test-TargetResource @testParams | Should Be $false
+            }
+
+            It "creates a new service application in the set method" {
+                Set-TargetResource @testParams
+                Assert-MockCalled New-SPAppManagementServiceApplication
+            }
+        }
+
+        Context "When service applications exist in the current farm but the specific Add-in app does not" {
+
+            Mock Get-SPServiceApplication { return @(@{
+                TypeName = "Some other service app type"
+            }) }
+
+            It "returns null from the Get method" {
+                Get-TargetResource @testParams | Should BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+        }
+
+        Context "When a service application exists and is configured correctly" {
+            Mock Get-SPServiceApplication { 
+                return @(@{
+                    TypeName = "App Management Service Application"
+                    DisplayName = $testParams.Name
+                    ApplicationPool = @{ Name = $testParams.ApplicationPool }
+                    Database = @{
+                        Name = $testParams.DatabaseName
+                        Server = @{ Name = $testParams.DatabaseServer }
+                    }
+                })
+            }
+
+            It "returns values from the get method" {
+                Get-TargetResource @testParams | Should Not BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+            It "returns true when the Test method is called" {
+                Test-TargetResource @testParams | Should Be $true
+            }
+        }
+
+        Context "When a service application exists and the app pool is not configured correctly" {
```

Comment:
Can you add a context to test the set method when you don't specify the two database parameters